src/clients/github.js

```diff
@@ -0,0 +1,280 @@
+import get from 'lodash/get';
+import GitHub from 'github-api';
+import isEmpty from 'lodash/isEmpty';
+import trim from 'lodash/trim';
+import performWithRetries from '../util/performWithRetries';
+import generatePreview from '../util/generatePreview';
+
+const _anonymousGithub = new GitHub({});
+const COMMIT_MESSAGE = 'Created using Popcode: https://popcode.org';
+const MASTER = 'master';
+const GH_PAGES = 'gh-pages';
+const NETWORK_ERROR = 'Network Error';
+const _parser = new DOMParser();
+
+function EmptyGistError(message) {
+  this.name = 'EmptyGistError';
+  this.message = message;
+  this.stack = new Error().stack;
+}
+EmptyGistError.prototype = Object.create(Error.prototype);
+
+function _getPageTitle(preview) {
+  return _parser.parseFromString(
+    preview,
+    'text/html').title.replace(/ /g, '-');
+}
+
+async function createRepoFromProject(project, user) {
+  const github = _clientForUser(user);
+  const preview = generatePreview(project);
+  const title = _getPageTitle(preview);
+
+  const REPO_NAME = `${title}-${Date.now()}`;
+
+  const {data} =
+    await _performWithRetryNotFoundOrNetworkErrors(
+      () => github.getUser().createRepo({name: REPO_NAME, auto_init: true}),
+    );
+
+  const userName = data.owner.login;
+
+  await _performWithRetryNotFoundOrNetworkErrors(
+    () => _createBranch(github, userName, REPO_NAME, MASTER, GH_PAGES),
```

Comment:
I couldn't figure out how to make it work, if you look at the source it's not really 'optional' in that you don't need an old branch, there's just a default if you don't fill it in: http://github-tools.github.io/github/docs/3.1.0/Repository.js.html#line514